dependencies.yaml

```diff
@@ -364,7 +392,7 @@ dependencies:
           # Sync with conda build constraint & wheel run constraint.
           # TODO: Change to `2.0.*` for NumPy 2
           - numpy==1.23.*
-  build_python_pylibcudf:
+  build_python_libcudf:
     common:
       - output_types: conda
         packages:
```

Comment:
(I can't comment on the line I want due to the diff placement.)

`build_python_libcudf` should probably have a `conda` dependency on `librmm` and not `rmm`. No Python bits from `rmm` should be needed to build a `libcudf` wheel in a conda env. (Or if we think the `conda` section is nonsensical to include for C++ wheel-builds, we can remove the `conda` section entirely.)